### PR TITLE
Support Enterprise Managed User in userGroup Regex

### PR DESCRIPTION
**Pull request by EvHaus**
### Initial checklist

* [x] I read the support docs 
* [x] I read the contributing guide 
* [x] I agree to follow the code of conduct 
* [x] I searched issues and discussions and couldn’t find anything or linked relevant results below 
* [x] I made sure the docs are up to date
* [x] I included tests (or that’s not needed)

### Description of changes

A GitHub Enterprise customer discovered a bug in our product today (that uses `remark-github`). Their code was crashing with:

```
Unexpected invalid `repository`, expected for example `user/project`
```

After a bunch of troubleshooting I discovered it's because their GitHub username has an `_` (underscore) in it. Turns out although generally this is not allowed on GitHub, Enterprise Managed Users can have underscores (see [official docs](https://docs.github.com/en/enterprise-cloud@latest/admin/managing-iam/iam-configuration-reference/username-considerations-for-external-authentication)).

This PR updates the Regex to support this and adds a unit test to catch this edge case.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -132,10 +132,11 @@ const denyHash = new Set([
 const minShaLength = 7
 
 // Username may only contain alphanumeric characters or single hyphens, and
-// cannot begin or end with a hyphen*.
+// cannot begin or end with a hyphen*. Enterprise Managed Users can have
+// underscores.
 //
 // \* That is: until <https://github.com/remarkjs/remark-github/issues/13>.
-const userGroup = '[\\da-z][-\\da-z]{0,38}'
+const userGroup = '[\\da-z](?:[-_\\da-z]{0,37}[\\da-z-])?'
```
Comment (EvHaus) [inline lib/index.js:139]: I'll be honest, GPT-5 did most of the work here as my Regex skills aren't good enough to optimize this. Two things needed:

1. Allow `_` in addition to `-` and `a-z`
2. BUT -- make sure `_` is not the last character in the string, otherwise this could break markdown formatting

**Comment by wooorm** [comment]
Do you have an exact quote from the website you are basing this on?
Or, an exact case of input?
From quickly glancing over that (sorry, little time), it seems to say that *one* underscore is used:

> On GitHub.com, GitHub also adds an underscore and your enterprise's shortcode to the end of each username.

> Usernames, including underscore and short code, must not exceed 39 characters.

Sounds like perhaps something else is needed, allowing one underscore + shortcode and checking length with code, rather than just allowing arbitrary underscores except at the last character? 🤔

**Comment by EvHaus** [comment]
> Do you have an exact quote from the website you are basing this on?

Yup -- it's that one that you found: "On GitHub.com, GitHub also adds an underscore and your enterprise's shortcode to the end of each username.".

> Or, an exact case of input?

The exact user id I ran into was `k-archarya_zse` where I think `_zse` is the GHC instance suffix.

> Sounds like perhaps something else is needed, allowing one underscore + shortcode and checking length with code, rather than just allowing arbitrary underscores except at the last character? 🤔

Sure. I can revise the Regex to only check for a single instance.
